draft-west-cookie-incrementalism.md

```diff
@@ -401,85 +404,281 @@ from "have the same registrable domain" (or "is an exact match for") to say
-     date and time. Set the scheme to request-uri's origin's scheme
-     component.
+3.  Create a new cookie with name cookie-name, value cookie-value. Set
+    the creation-time and the last-access-time to the current date and
+    time.
 ~~~
 
-Likewise alter step 17 of the same algorithm from:
+to: 
 
 ~~~
-17.  If the cookie store contains a cookie with the same name,
-     domain, host-only-flag, and path as the newly-created cookie:
+3.  Create a new cookie with name cookie-name, value cookie-value. Set
+    the creation-time and the last-access-time to the current date and
+    time, set the scheme to the request-uri's origin's scheme
+    component, and set the port to the request-uri's origin's port
+    component.
 ~~~
 
-to:
+Because all cookies are now scoped by scheme it is unnecessary to
+protect a secure scheme's cookies from an insecure scheme's cookies and
+thus we can remove step 13.
+
+To incorporate the new fields and "domain cookie" port matching, alter
+step 19 of the same algorithm from
+
+~~~
+19. If the cookie store contains a cookie with the same name, domain,
+    host-only-flag, and path as the newly-created cookie:
+
+    1. Let old-cookie be the existing cookie with the same name,
+       domain, host-only-flag, and path as the newly-created cookie.
+       (Notice that this algorithm maintains the invariant that there
+       is at most one such cookie.)
+
+    2. If the newly-created cookie was received from a "non-HTTP" API
+       and the old-cookie's http-only-flag is true, abort these steps
+       and ignore the newly created cookie entirely.
+
+    3. Update the creation-time of the newly-created cookie to match
+       the creation-time of the old-cookie.
 
+    4. Remove the old-cookie from the cookie store.
 ~~~
-17.  If the cookie store contains a cookie with the same name, scheme,
-     domain, host-only-flag, and path as the newly-created cookie:
+
+to:
+
 ~~~
+19. If the cookie store contains a cookie ("old-cookie") with the same
+    name, scheme, domain, host-only-flag, path, and for which
+    the old-cookie's port port-matches the newly-created cookie:
 
-And step 17.1 from:
+    1. If the newly-created cookie was received from a "non-HTTP" API
+       and the old-cookie's http-only-flag is true, abort these steps
+       and ignore the newly created cookie entirely.
 
+    2. Update the creation-time of the newly-created cookie to match
+       the creation-time of the old-cookie.
+
+    3. Remove the old-cookie from the cookie store.
+    (Notice that this algorithm maintains the invariant that there
+    is at most one such cookie.)
 ~~~
-17.1.  Let old-cookie be the existing cookie with the same name,
-       domain, host-only-flag, and path as the newly-created
-       cookie.  (Notice that this algorithm maintains the invariant
-       that there is at most one such cookie.)
+
+With cookie storage taken care of now we move onto sending cookies. In
+section 5.5 alter step 1 from:
```

Comment:
```suggestion
section 5.6.1 alter step 1 from:
```

Also: step 1 contains more conditions than those listed here. It might be clearer to "replace step 1 with the following", and include all of them (`http-only-flag` and `same-site-flag`, at least, are currently missing).